src/wallet/rpcdump.cpp

```diff
@@ -550,6 +550,48 @@ UniValue dumpprivkey(const JSONRPCRequest& request)
     return CBitcoinSecret(vchSecret).ToString();
 }
 
+UniValue dumpmasterprivkey(const JSONRPCRequest& request)
+{
+    if (!EnsureWalletIsAvailable(request.fHelp))
+        return NullUniValue;
+    
+    if (request.fHelp || request.params.size() != 0)
+        throw runtime_error(
+            "dumpmasterprivkey\n"
+            "\nReveals the current master private key.\n"
+            "\nResult:\n"
+            " \"key\"                 (string) The HD master private key\n"
+            "\nExamples:\n"
+            + HelpExampleCli("dumpmasterprivkey", "")
+        );
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    EnsureWalletIsUnlocked();
+
+    CKeyID masterKeyID = pwalletMain->GetHDChain().masterKeyID;
```

Comment:
Well,... you need the masterKeys `CKeyID` anyway,... but I guess it would still make sense to use `CWallet::IsHDEnabled()`